src/vm/contexts.rs

```diff
@@ -779,6 +799,23 @@ impl<'a, 'b> Environment<'a, 'b> {
         result
     }
 
+    /// Used only for contract-call! cost short-circuiting. Once the short-circuited cost
+    ///  has been evaluated and assessed, the contract-call! itself is executed "for free".
+    pub fn run_free<F, A>(&mut self, to_run: F) -> A
+    where
+        F: FnOnce(&mut Environment) -> A,
+    {
+        let original_tracker = replace(
+            &mut self.global_context.cost_track,
+            LimitedCostTracker::new_free(),
+        );
+        // note: it is important that this method not return until original_tracker has been
+        //  restored. DO NOT use the try syntax (?).
```

Comment:
Not sure what this comment is saying -- isn't that already guaranteed here by the way the code is written?  Or, is this a comment on the implementation of `to_run`?

Reply:
Yeah, it's a note to future modifiers of that code.